src/agent/agent.c

```diff
@@ -1029,6 +1029,16 @@ bool agent_start(Agent *agent) {
                 return false;
         }
 
+        r = bus_socket_set_no_delay(agent->peer_dbus);
+        if (r < 0) {
+                hirte_log_warn("Failed to set NO_DELAY on socket");
+        }
+
+        r = bus_socket_set_no_delay(agent->peer_dbus);
```

Comment:
This looks like you have a cut and paste error? You set no delay twice?

Reply:
```suggestion
        r = bus_socket_set_keepalive(agent->peer_dbus);
```